src/plugins/cpptemplate/cpptemplate_delegate.cpp

```diff
@@ -0,0 +1,24 @@
+/**
+ * @file
+ *
+ * @brief Delegate implementation for the `cpptemplate` plugin
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include "cpptemplate_delegate.hpp"
+
+namespace elektra
+{
+using CppKeySet = kdb::KeySet;
+
+/**
+ * @brief This constructor creates a new delegate object used by the `cpptemplate` plugin
+ *
+ * @param config This key set contains configuration values provided by the `cpptemplate` plugin
+ */
+CppTemplateDelegate::CppTemplateDelegate (CppKeySet config ELEKTRA_UNUSED)
+{
+}
```

Comment:
open, close, ... not implemented?

Reply:
The `open` and `close` functions are implemented in the delegator ([`kdbplugin.hpp`](https://github.com/ElektraInitiative/libelektra/blob/master/src/bindings/cpp/include/kdbplugin.hpp)).

Reply:
Maybe some comment like "implement open functionality here" would help?

Btw. do you really like the style that everything starts with Cpp? Would you call a ruby function Ruby...?

Reply:
> Btw. do you really like the style that everything starts with Cpp?

Not, really. However since `Key`/`KeySet` is usually already reserved for `ckdb::Key`/ `ckdb::KeySet` I tend to just add the `Cpp` prefix. If you have a better naming option in mind, I am open for suggestions. For now I changed the name to the usual `KeySet` in the delegate.

> Would you call a ruby function Ruby...?

Not really, but usually I do not encounter naming conflicts in languages such as Python or Ruby.